validations/builder_schema.go

```diff
@@ -7,45 +7,46 @@ import (
 	"github.com/SAP/cloud-mta/mta"
 )
 
-func checkStringParam(params map[string]interface{}, parentNode *yaml.Node, paramsListName string, paramName string) []YamlValidationIssue {
-	var issues []YamlValidationIssue
+func checkStringProperty(props map[string]interface{}, propsNode *yaml.Node, propName string) []YamlValidationIssue {
 
-	if params != nil && params[paramName] != nil {
-		paramsNode := getPropValueByName(parentNode, paramsListName)
-		_, ok := params[paramName].(string)
-		if !ok {
-			paramNode := getPropValueByName(paramsNode, paramName)
-			issues = appendIssue(issues,
-				fmt.Sprintf(`the "%s" property is defined incorrectly; the property must be a string`, paramName),
-				paramNode.Line)
+	_, ok := props[propName].(string)
+	if props != nil && props[propName] != nil && !ok {
+		propNode := getPropValueByName(propsNode, propName)
+		return []YamlValidationIssue{
+			{
+				Msg:  fmt.Sprintf(`the "%s" property is defined incorrectly; the property must be a string`, propName),
+				Line: propNode.Line,
+			},
 		}
 	}
-	return issues
+	return nil
 }
 
 func checkBuilderSchema(mta *mta.MTA, mtaNode *yaml.Node, source string) []YamlValidationIssue {
 	var issues []YamlValidationIssue
 
-	issues = append(issues, checkStringParam(mta.Parameters, mtaNode, parametersYamlField, deployModeYamlField)...)
+	issues = append(issues, checkStringProperty(mta.Parameters, getPropValueByName(mtaNode, parametersYamlField), deployModeYamlField)...)
```

Comment:
Won't this add nil to the issues slice if there's no error?

Reply:
no, it receives slice, that can be empty or nil. "a nil slice is functionally equivalent to a zero-length slice, even though it points to nothing"  (from: https://blog.golang.org/slices). It works in both directions: we can append to nil slice and we can append a nil slice.